keep/actions/actions_factory.py

```diff
@@ -0,0 +1,82 @@
+import time
+import logging
+import yaml
+from io import StringIO
+from uuid import uuid4
+from typing import List, Union
+from keep.api.models.action import ActionDTO
+from keep.api.models.db.action import Action
+from keep.api.core.db import get_all_actions, create_actions, delete_action, get_action, update_action
+from keep.actions.actions_exception import ActionsFactoryException
+
+logger = logging.getLogger(__name__)
+
+class ActionsFactory:
+    """CRUD for Action model that shares across CLI, API, ..."""
+    @staticmethod
+    def get_all_actions(tenant_id: str) -> List[ActionDTO]:
+        action_models = get_all_actions(tenant_id)
+        action_dtos = map(ActionsFactory._convert_model_to_dto, action_models)
```

Comment:
one wrong action will return 500 for all others. should be just logs about the broken action and return the others

Reply:
Revised to to log warning of missed record and return the others as follows. 

```
@staticmethod
    def get_all_actions(tenant_id: str) -> List[ActionDTO]:
        action_models = get_all_actions(tenant_id)
        return ActionsCRUD._convert_models_to_dtos(action_models)

    @staticmethod
    def _convert_models_to_dtos(models: List[Action]) -> List[ActionDTO]:
        """Convert model to dto, ignore the result if one model is invalid"""
        results: List[ActionDTO] = []
        for model in models:
            try:
                dto = ActionDTO(id=model.id, use=model.use, name=model.name, details=yaml.safe_load(StringIO(model.action_raw)))
                results.append(dto)
            except ValidationError:
                logger.warning("Unmatched Action model and the coresponding DTO", exc_info=True, extra={
                    "data": model.dict()
                })
        return results
```